# Following patch was applied to STM32U5's HAL OSPI driver to disable optimizations
# for certain functions. Compiling the app in -Os mode for certain STM32CubeIDE
# versions causes the driver to not function correctly.

diff --git a/Drivers/STM32U5xx_HAL_Driver/Src/stm32u5xx_hal_ospi.c b/Drivers/STM32U5xx_HAL_Driver/Src/stm32u5xx_hal_ospi.c
index 3d38b50..22643c7 100644
--- a/Drivers/STM32U5xx_HAL_Driver/Src/stm32u5xx_hal_ospi.c
+++ b/Drivers/STM32U5xx_HAL_Driver/Src/stm32u5xx_hal_ospi.c
@@ -3000,6 +3000,7 @@ static void OSPI_DMAAbortCplt(DMA_HandleTypeDef *hdma)
   * @param  Tickstart : Tick start value
   * @retval HAL status
   */
+__attribute__((optimize("O0")))
 static HAL_StatusTypeDef OSPI_WaitFlagStateUntilTimeout(OSPI_HandleTypeDef *hospi, uint32_t Flag,
                                                         FlagStatus State, uint32_t Tickstart, uint32_t Timeout)
 {
@@ -3031,6 +3032,7 @@ static HAL_StatusTypeDef OSPI_WaitFlagStateUntilTimeout(OSPI_HandleTypeDef *hosp
   * @param  cmd   : structure that contains the command configuration information
   * @retval HAL status
   */
+__attribute__((optimize("O0")))
 static HAL_StatusTypeDef OSPI_ConfigCmd(OSPI_HandleTypeDef *hospi, OSPI_RegularCmdTypeDef *cmd)
 {
   HAL_StatusTypeDef status = HAL_OK;
